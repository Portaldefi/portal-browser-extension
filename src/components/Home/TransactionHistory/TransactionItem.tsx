import React from 'react';
import { Checkbox, Grid, Header } from 'semantic-ui-react';

interface ITransactionItem {
  tx: any
};

const shortenString = (str: string) => {
  return str.substring(0, 6) + '...' + str.substring(str.length - 3);
};

const dateToString = (dat: Date) => {
  return dat.getFullYear() + '-' + (dat.getMonth() + 1) + '-' + dat.getDate();
}

export default ({tx}: ITransactionItem) => {
  console.log(tx);
  return (
    <Grid className='p-16 py-9 bg-white'>
      <Grid.Column width='eleven' textAlign='left'>
        <Header as='p' className='description'>{shortenString(tx.txid)}</Header>
        <Header as='p' className='description'>{dateToString(new Date(tx.status.block_time))}</Header>
      </Grid.Column>
      <Grid.Column width='five' verticalAlign='middle'>
        <Header as='p' className='description'>{tx.status.confirmed === true ? "Confirmed" : "Unconfirmed"}</Header>
        <Header as='p' className='description'>{tx.vout[1].value - tx.vout[0].value}</Header>
      </Grid.Column>
    </Grid>
  )
}