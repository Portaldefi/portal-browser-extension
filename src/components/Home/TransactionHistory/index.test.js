import Connectionitem from './ConnectionItem';
import TransactionItem from './TransactionItem';
import { create, act } from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

describe('TransactionHistory', () => {
  const initialState = {
    menu: {
      items: [],
      transactions: [{
        txid: '',
        status: {
          block_time: '',
          confirmed: true,
        },
        vout: [{value: 0}]
      }]
    },
    key: {
      privateKey: '',
      privateExtendedKey: '',
      identity: [
        [{
          address: 'test address'
        }]
      ],
      selectedIdentityId: 0,
      selectedChainId: 0
    }
  };
  const mockStore = configureStore();
  let store;
  it('ConnectionItem', () => {
    let tree;
    act(() => {
      tree = create(
        <Connectionitem />
      );
    });

    expect(tree).toMatchSnapshot();
  });
  it('TransactionItem', () => {
    let tree;
    act(() => {
      store = mockStore(initialState);
      tree = create(
        <Provider store={store}>
          <BrowserRouter>
            <TransactionItem id={0}/>
          </BrowserRouter>
        </Provider>
      );
    });

    expect(tree).toMatchSnapshot();
  });
  afterAll(() => jest.resetModules());
});