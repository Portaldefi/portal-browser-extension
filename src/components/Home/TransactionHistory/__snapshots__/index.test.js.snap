// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionHistory ConnectionItem 1`] = `
<div
  className="ui grid p-16 py-9 bg-white"
>
  <div
    className="left aligned eleven wide column"
  >
    <p
      className="ui header description"
    />
    <p
      className="ui header comment"
    />
  </div>
  <div
    className="middle aligned five wide column"
  >
    <div
      className="ui fitted toggle checkbox"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        readOnly={true}
        tabIndex={0}
        type="checkbox"
      />
      <label />
    </div>
  </div>
</div>
`;

exports[`TransactionHistory TransactionItem 1`] = `
<div
  className="ui grid p-16 py-9 bg-white"
  onClick={[Function]}
>
  <div
    className="left aligned eleven wide column"
  >
    <p
      className="ui header description"
    />
    <p
      className="ui header description"
    >
      NaN-NaN-NaN
    </p>
  </div>
  <div
    className="middle aligned five wide column"
  >
    <p
      className="ui header description"
    >
      Confirmed
    </p>
    <p
      className="ui header description"
    >
      0
    </p>
  </div>
</div>
`;
